joomla/joomla-cms
# Implement Session GC Cli

Pull Request for Issue # .

### Summary of Changes
Since Joomla 3.8.4, if you use Database Session Handler, session data won't be deleted on page loads like in older version of Joomla anymore, it now relies on PHP session data garbage collection proces to delete session data and could cause session data stay longer in database than expected. On some hostings, session.gc_probability http://php.net/manual/en/session.configuration.php#ini.session.gc-probability even has value set to 0, mean session data won't be deleted at all.

This PR implements a simple CLI script so that you can setup cron job to clean up session data if needed. It will be more reliable than rely on session data garbage collection and for high traffic website, you will get big performance improvement (set session.gc_probability  to 0 and setup cron job to delete session data- see https://github.com/joomla/joomla-cms/issues/19146 for a use case)

### Testing Instructions
1. Go to System - Global Configuration, look at System tab, set Session Handler to Database, Session Lifetime to 1 minute

2. Access to frontend of your site so that a new record will be inserted into #__session table

3. Access to your site database via PHPMyadmin, look at #__session, you should see some records

4. Wait for about one minute

5. Run this CLI script. I don't know about Linux, but on Windows, I just need to run this command from Command Line **php D:\www\joomla\cli\sessionGc.php**  (D:\www\joomla needs to be replaced with the path of your Joomla installation)

6. Check #__session table again, you should see that the old records were deleted (maybe all records deleted)

### Documentation Changes Required
Maybe we need to document this new feature if accepted
